### PR TITLE
Replacing .NET regex for RE2 regex

### PR DESCRIPTION
**Pull request by eddynaka**

### CODE REVIEW

File: Src/RE2.Managed/RE2Regex.cs [hunk 1]
```diff
@@ -15,7 +14,9 @@ public class RE2Regex : IRegex
         public static IRegex Instance = new RE2Regex();
         public static TimeSpan DefaultTimeout = TimeSpan.FromMilliseconds(int.MaxValue - 1);
 
-        private RE2Regex() { }
+        private RE2Regex()
+        {
+        }
```
Comment (v-kevinvu) [inline Src/RE2.Managed/RE2Regex.cs:19]: Is this needed?

File: Src/Sarif.PatternMatcher/CachedDotNetRegex.cs [hunk 2]
```diff
@@ -27,7 +27,9 @@ static CachedDotNetRegex()
             RegexCache = new ConcurrentDictionary<Tuple<string, RegexOptions>, Regex>();
         }
 
-        private CachedDotNetRegex() { }
+        private CachedDotNetRegex()
+        {
+        }
```
Comment (v-kevinvu) [inline Src/Sarif.PatternMatcher/CachedDotNetRegex.cs:32]: Is this needed?

File: Src/Sarif.PatternMatcher/DotNetRegex.cs [hunk 3]
```diff
@@ -19,7 +19,9 @@ public class DotNetRegex : IRegex
 
         internal static TimeSpan DefaultTimeout = TimeSpan.FromMilliseconds(int.MaxValue - 1);
 
-        private DotNetRegex() { }
+        private DotNetRegex()
+        {
+        }
```
Comment (v-kevinvu) [inline Src/Sarif.PatternMatcher/DotNetRegex.cs:24]: Is this needed?

**Review by v-kevinvu — DISMISSED** [review]
:shipit:

File: Src/RE2.Managed/RE2Regex.cs (hunk 1 above)
Comment (eddynaka) [reply Src/RE2.Managed/RE2Regex.cs:19]: my format automatically makes this change. will revert later.

File: Src/RE2.Managed/Regex2.cs [hunk 4]
```diff
@@ -273,14 +273,16 @@ public static unsafe bool Matches(string pattern, string text, out List<Dictiona
                                 submatchString = Encoding.UTF8.GetString(textUtf8Bytes, submatchTextStartIndex, submatchLength);
                             }
 
+                            var flexMatch = new FlexMatch() { Success = true, Index = submatchTextStartIndex, Length = submatchLength, Value = submatchString };
```
Comment (v-kevinvu) [inline Src/RE2.Managed/Regex2.cs:276]: >submatchLength [](http://example.com/codeflow?start=119&length=14)

`submatchLength` != `submatchString.Length`

Use `submatchString.Length` here.

**Review by v-kevinvu — CHANGES_REQUESTED** [review]
:clock1:

File: Src/RE2.Managed/Regex2.cs (hunk 4 above)
Comment (v-kevinvu) [inline Src/RE2.Managed/Regex2.cs:276]: >submatchTextStartIndex [](http://example.com/codeflow?start=86&length=22)

This is the start index in a byte buffer which represents a UTF-8 encoded string.

It is not necessarily the same as the start index of the Unicode character in a Unicode string.

**Pull request merged**